### PR TITLE
fix: Bug archived read only mode variants are editable even tho not persisted

### PR DESCRIPTION
**Pull request by aiAdrian**
# Description

When the project is archived, the user can no longer modify the netzgrafik. All editing functions should be disabled.

# Testing

This bug fix is a big one. Therefore, we have to deeply test the user interface interactions. There should be no differences from the current version as long as the project is not archived. If a project along with their variants are archived, the display of these variants is read-only. That means the user can no longer make any changes.

* [ ] All manual tests passed
 

# Checklist

* [x] This PR contains a description of the changes I'm making
* [x] I've read the [Contribution Guidelines](https://github.com/SchweizerischeBundesbahnen/netzgrafik-editor-frontend/blob/main/CONTRIBUTING.md)
* [x] I've added tests for changes or features I've introduced
* [x] I documented any high-level concepts I'm introducing in `documentation/`
* [x] CI is currently green and this is ready for review

### CODE REVIEW

**Comment by aiAdrian** [comment]
> I will take a close look to this PR. Have you seen this related issue ? #191 The issue is that NGE integrated version currently runs with the bugged `read-only` mode (so yet not fixed by this PR). By merging this PR, the integrated versions of NGE will be broken. Had you have the time to think about a 3rd mode yet ?

Thank you for your comment. I believe integrating the standalone application will be quite simple. If environment.backendDisabled is set to true, the getVariantIsWritable method in VersionControlService should always return true. Additionally, the environment object could be extended with a flag - readonly: boolean. If readonly is set to true, everything can be handled in VersionControlService to enforce the read-only mode. Possibly, this method should be moved to another service, which requires a software architectural review that has not yet been done in detail. However, this can be easily achieved through refactoring.

**Comment by louisgreiner** [comment]
image

It would be nice to hide this grey bar on bottom on button where the mouse is (hover) ? Disable this bar when a button is disable

**Comment by aiAdrian** [comment]
I have overworked the styling and the behavior (only styling) when:
- readonly
- if Streckengrafik is on
- if Analytics is on
for the menu. It should now be finished/polished. But please re-test. Thanks.

**Comment by louisgreiner** [comment]
Shouldn't archived projects **automatically** archive variants that aren't archived yet ?

image

Here, the variant isn't archived, even tho the project is, but the if we open the variant, the mode is archived

image

The solution would be to automatically archive all the variants of a project that is archived, and tick and disable by default the Show archive button ?

image

File: src/app/perlenkette/perlenkette-node/perlenkette-node.component.html [hunk 1]
```diff
@@ -1,4 +1,6 @@
-<div class="node-container" (click)="disapleSectionView($event)">
+<div class="node-container" (click)="disapleSectionView($event)"
```
Comment (SergeCroise) [inline src/app/perlenkette/perlenkette-node/perlenkette-node.component.html:1]:
```suggestion
<div class="node-container" (click)="disableSectionView($event)"
```

**Comment by aiAdrian** [comment]
> Shouldn't archived projects **automatically** archive variants that aren't archived yet ?
> 
> image
> 
> Here, the variant isn't archived, even tho the project is, but the if we open the variant, the mode is archived
> 
> image
> 
> The solution would be to automatically archive all the variants of a project that is archived, and tick and disable by default the Show archive button ?
> 
> image

I'm not sure if I understood everything correctly. So I would like to briefly explain what the archive concept looks like:

If a project is archived, then all attached (assigned) variants are read-only. (Here I have just identified a problem, i.e., if a project is archived, there is an auth error if one tries to archive a non-archived variant -> read-only protection applies). However, the variants are not shown as archived.

If a project is not archived, variants can be archived, as the project is not read-only. However, this requires that the user has write permissions.

Archiving means deleted – in other words, only archived variants can be deleted. The same applies to projects. It’s a two-step process, so that on one hand, the user can be really sure that they want to delete something. Or one can define in the process that all work should remain saved, then everything that is no longer needed is archived and that’s it.

**Comment by aiAdrian** [comment]
> > Shouldn't archived projects **automatically** archive variants that aren't archived yet ?
> > ...
> > The solution would be to automatically archive all the variants of a project that is archived, and tick and disable by default the Show archive button ?
> 
> I'm not sure if I understood everything correctly. So I would like to briefly explain what the archive concept looks like:
> 
> If a project is archived, then all attached (assigned) variants are read-only. (Here I have just identified a problem, i.e., if a project is archived, there **is an auth error if one tries to archive** a non-archived variant -> read-only protection applies). However, the variants are not shown as archived.
> 
> ...

The above reported ***error fixed*** is fixed : https://github.com/SchweizerischeBundesbahnen/netzgrafik-editor-frontend/pull/257/commits/4b6c5e52dc8680f973f5726eeac355ffd2e96167

File: src/app/perlenkette/perlenkette-node/perlenkette-node.component.html (hunk 1 above)
Comment (aiAdrian) [reply src/app/perlenkette/perlenkette-node/perlenkette-node.component.html]: Should be fixed. Thanks

File: src/app/view/editor-main-view/data-views/trainrunsections.view.spec.ts [hunk 2]
```diff
@@ -171,7 +171,8 @@ describe("TrainrunSection-View", () => {
       copyService,
       logService,
       viewportCullSerivce,
```
Comment (SergeCroise) [inline src/app/view/editor-main-view/data-views/trainrunsections.view.spec.ts:173]: ?

**Comment by louisgreiner** [comment]
Thanks for your explanations in [your previous comment](https://github.com/SchweizerischeBundesbahnen/netzgrafik-editor-frontend/pull/257#issuecomment-2311650809), I better understand the logic : the variants inside an archived project should not be able to be (un)archived afterwards. The user can then see the archived and unarchived variants of an archived project -> working well

**Review by louisgreiner — APPROVED** [review]

**Pull request merged**